common/src/Utilities/PString.cpp

```diff
@@ -0,0 +1,95 @@
+#include "PrecompiledHeader.h"
+#include "PString.h"
+#include <codecvt>
+#include <wchar.h>
+
+const PString& PString::operator=(const PString& rhs)
+{
+	string = rhs.string;
+	return *this;
+}
+
+PString& PString::operator=(PString&& rhs)
+{
+	string = std::move(rhs.string);
+	return *this;
+}
+
+#ifdef _WIN32
+// Non default
+PString::PString(const std::wstring& str)
+{
+	char* temp = new char[count];
```

Comment:
`std::string` has a non-const `data()` overload now, so this can be done in-place with a properly done `.resize`.

Reply:
I was having issues with data() and wcstombs in this functions it didn't seem to want to fill it in proper

Reply:
You shouldn't use `wcstombs` in the first place, maybe that was the issue you encountered in the first place.